### PR TITLE
Workaround SWDEV_247082

### PR DESCRIPTION
**Pull request by zjing14**
- Disabled kperblock = 8

@asroy update:
This is to workaround a compiler bug
http://ontrack-internal.amd.com/browse/SWDEV-247082

The failure was on 3.5/3.6
```
bin/MIOpenDriver conv -n 64 -c 16 -k 96 -H 55 -W 55 -y 1 -x 1 -u 2 -v 2 -p 0 -q 0 -g 1 -i 1 -F 2
```

### CODE REVIEW

**Comment by daniellowell** [comment]
@zjing14 What is going on with this PR?

**Comment by zjing14** [comment]
The PR is for temporarily workaround the failure reported in SWDEV_247082. The PR is ready.

**Comment by atamazov** [comment]
It looks good to me. Needs to push it through CI.

**Comment by atamazov** [comment]
Let me push it once...

**Comment by atamazov** [comment]
Restarted CI job another time. /cc @zjing14

File: src/solver/implicitgemm_util.hpp [hunk 1]
```diff
@@ -23,6 +23,7 @@ MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_BLOCK_SYNC_LDS_WITHOUT_SY
 // LLVM xdlops instrinsic will do unnecessey VGRP <--> AGPR movement, and result in
 // register spill, for bfloat16 datatype, when doing wave-wise GEMM larger than 64x64
 #define WORKAROUND_SWDEV_240356 1
+#define WORKAROUND_SWDEV_247082 1
```
Comment (asroy) [inline src/solver/implicitgemm_util.hpp:26]: Please add comment describing the issue a bit.

**Review by asroy — CHANGES_REQUESTED** [review]

**Comment by asroy** [comment]
@zjing14 Is the bug only to fp32 or fp16/bfp16 as well?

File: src/solver/implicitgemm_util.hpp [hunk 2]
```diff
@@ -516,6 +517,11 @@ static inline bool IsValidBlockwiseGemmXdlops(const ConvolutionContext& ctx,
                                               const int GemmNPerWave,
                                               const int GemmKPack)
 {
+#if WORKAROUND_SWDEV_247082
+    if(GemmKPerBlock == 8)
```
Comment (asroy) [inline src/solver/implicitgemm_util.hpp:521]: Does this bug happen for all data type?

Comment (zjing14) [reply src/solver/implicitgemm_util.hpp:521]: At least, happened for fp32/fp16.

**Comment by daniellowell** [comment]
@zjing14 Merge conflicts.

**Comment by zjing14** [comment]
@daniellowell PR https://github.com/ROCmSoftwarePlatform/MIOpen/pull/423 with the same workaround already merged. Closed this.

**Pull request closed without merging**